ainstein_radar_drivers/src/radar_interface_o79_udp.cpp

```diff
@@ -171,32 +155,41 @@ void RadarInterfaceO79UDP::mainLoop(void)
 		  ainstein_radar_filters::data_conversions::radarTargetArrayToROSCloud( *radar_data_msg_ptr_raw_, *cloud_msg_ptr_raw_ );
 		  pub_cloud_raw_.publish( cloud_msg_ptr_raw_ );
 		}
-
 	    }
 
 	  if( targets_tracked.size() > 0 )
 	    {
-	      // Fill in the tracked RadarTargetArray message from the received targets:
+      	      // Fill in the tracked RadarTargetArray message from the received targets:
 	      radar_data_msg_ptr_tracked_->header.stamp = ros::Time::now();
-	      radar_data_msg_ptr_tracked_->targets.clear();
+	      radar_data_msg_ptr_tracked_->objects.clear();
+	      
+	      ainstein_radar_msgs::RadarTrackedObject obj;
 	      for( const auto &t : targets_tracked )
 		{
-			if (t.id >= 0)
-			{
-				radar_data_msg_ptr_tracked_->targets.push_back( targetToROSMsg( t ) );
-			}
+		  obj = utilities::targetToObjectROSMsg( t );
```

Comment:
still need to check for t.id >= 0 here. When I push back an empty target, I give it id=-1 so that I can look for that condition here to not populate the range, velocity, etc. If we don't do this then it looks like there's a target at the origin in rviz

Reply:
Good point - I think I removed this because I thought it was causing a different issue I was facing at the time. I added the check back in.